From d65382827076c81519afbf43cfb5ca071a990c14 Mon Sep 17 00:00:00 2001
From: Scibian Packaging Team <packaging@scibian.org>
Date: Mon, 22 Aug 2016 16:46:34 +0200
Subject: [PATCH] D-Bus: allow setting empty network properties

In certain situations, it permit relying on key auto detection by setting an
empty key_id property, for example.
---
 wpa_supplicant/dbus/dbus_new_handlers.c | 17 ++++++++++-------
 1 file changed, 10 insertions(+), 7 deletions(-)

diff --git a/wpa_supplicant/dbus/dbus_new_handlers.c b/wpa_supplicant/dbus/dbus_new_handlers.c
index 9f6c4a3..cc54173 100644
--- a/wpa_supplicant/dbus/dbus_new_handlers.c
+++ b/wpa_supplicant/dbus/dbus_new_handlers.c
@@ -207,7 +207,7 @@ dbus_bool_t set_network_properties(struct wpa_supplicant *wpa_s,
 		value = NULL;
 		if (entry.type == DBUS_TYPE_ARRAY &&
 		    entry.array_type == DBUS_TYPE_BYTE) {
-			if (entry.array_len <= 0)
+			if (entry.array_len < 0)
 				goto error;
 
 			size = entry.array_len * 2 + 1;
@@ -215,15 +215,18 @@ dbus_bool_t set_network_properties(struct wpa_supplicant *wpa_s,
 			if (value == NULL)
 				goto error;
 
-			ret = wpa_snprintf_hex(value, size,
-					       (u8 *) entry.bytearray_value,
-					       entry.array_len);
-			if (ret <= 0)
-				goto error;
+			if(entry.array_len > 0)
+			{
+				ret = wpa_snprintf_hex(value, size,
+						       (u8 *) entry.bytearray_value,
+						       entry.array_len);
+				if (ret <= 0)
+					goto error;
+			}
 		} else if (entry.type == DBUS_TYPE_STRING) {
 			if (should_quote_opt(entry.key)) {
 				size = os_strlen(entry.str_value);
-				if (size <= 0)
+				if (size < 0)
 					goto error;
 
 				size += 3;
-- 
2.1.4

